From ef945b41048e56c9e952c3582051d99a46095352 Mon Sep 17 00:00:00 2001
From: Rouven Czerwinski <r.czerwinski@pengutronix.de>
Date: Thu, 24 Mar 2022 10:04:31 +0100
Subject: [PATCH] imx8-media-dev: fix source sink pad setup

The NXP media driver assumes that there are only sensors connected to
the CSI interface. However for the bridge we don't want to connect to
ep0 which is the sink for the parallel input. Hard code it to source_pad
1 which is the sink port of the MIPI CSI output.

Signed-off-by: Rouven Czerwinski <r.czerwinski@pengutronix.de>
Signed-off-by: Ian Dannapel <ian.dannapel@iris-sensing.com>
Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 drivers/staging/media/imx/imx8-media-dev.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/media/imx/imx8-media-dev.c b/drivers/staging/media/imx/imx8-media-dev.c
index 534fa5a4cd14..7c1ae75cb2f0 100644
--- a/drivers/staging/media/imx/imx8-media-dev.c
+++ b/drivers/staging/media/imx/imx8-media-dev.c
@@ -519,8 +519,8 @@ static int mxc_md_create_links(struct mxc_md *mxc_md)
 
 			source = &sensor->sd->entity;
 			sink = find_entity_by_name(mxc_md, mipi_csi2->sd_name);
-			source_pad = 0;
-			sink_pad = source_pad;
+			source_pad = 1;
+			sink_pad = 0;
 
 			mipi_vc = (mipi_csi2->vchannel) ? 4 : 1;
 			for (j = 0; j < mipi_vc; j++) {
-- 
2.46.0

